### PR TITLE
Weblitcircle scott

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

File: less/pages/web-literacy.less [hunk 1]
```diff
@@ -44,19 +76,40 @@
...
   .weblit-nav {
-    display: flex;
-    justify-content: space-around;
-    align-items: center;
-
-    @problemsolving: #F1E5F0;
-    @communication: #FEFAE1;
-    @collaboration: #EEF5E8;
-    @creativity: #E2F3FA;
+    @problemsolving: #793575;
```
Comment (Pomax) [inline less/pages/web-literacy.less]: we probably want to base these either off of colours in the `variables.less` file (which houses all the other colours) with optional darken/ligten/saturate/etc mixins over them or just straight up move them there instead

File: lib/routes.jsx [hunk 2]
```diff
@@ -51,6 +50,25 @@ var redirects = {
 
 // aggregate all paths used in the app
 var urls = ['/'];
+urls = urls.concat(['web-literacy',
```
Comment (Pomax) [inline lib/routes.jsx]: just for neatness, can we put this first item on the next line, too?

File: lib/routes.jsx [hunk 3]
```diff
@@ -51,6 +50,25 @@ var redirects = {
...
+  'web-literacy/Write/Design',
+  'web-literacy/Write/Code',
+  'web-literacy/Write/Compose',
+  'web-literacy/Write/Revise',
+  'web-literacy/Write/Remix',
+  'web-literacy/Read',
+  'web-literacy/Read/Search',
+  'web-literacy/Read/Navigate',
+  'web-literacy/Read/Synthesize',
+  'web-literacy/Read/Evaluate']);
```
Comment (Pomax) [inline lib/routes.jsx]: and then put `]);` on the next line, similar to how you'd write objects

File: pages/web-literacy/MakeLinksFromWebLitSkills.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+var webLitSkillsContent = weblitcontent.webLitSkills;
+var ReactRouter = require('react-router');
+var Link = ReactRouter.Link;
+
+module.exports = function(webLitSkills) {
+  return (
+    <span>
+    {
+      webLitSkills.map(function(webLitSkill, index) {
+        var comma = "";
```
Comment (Pomax) [inline pages/web-literacy/MakeLinksFromWebLitSkills.jsx]: this feels like a thing we can do with CSS instead, so we don't use the `<span>` wrapper, we just return `<Link>` and then we use CSS to do:
```
.linkclass + .linkclass:after { content: ", "; }
.linkclass + .linkclass:last-child:after { content: ''; }
```
or something?

File: pages/web-literacy/Skills.jsx [hunk 5]
```diff
@@ -20,22 +21,47 @@ module.exports = React.createClass({
...
+                    src1x={skill.imgSrc1x}
+                    src2x={skill.imgSrc2x}
                     alt="">
-                    <h2>{skill.name}</h2>
+                    <h2 id={skill.name}>{skill.name}</h2>
                     <p>{skill.content}</p>
+                    <h3>Competencies</h3>
+                    <div className="competency-content">
+                      {
+                        skill.competencies.map(function(competency) {
```
Comment (Pomax) [inline pages/web-literacy/Skills.jsx]: given the double `map`, this feels like it should be
```
<div classname="...">
{ this.generateCompetencies() }
<//div>
```
and then in `generateCompetencies(skill)` you'd have
```
skill.competencies.map(function(competency) {
  return (
    <div key={competency.name}>
     <h4>{competency.name}</h4>
      { this.generateCompetency(competency) }
    </...>
  );
});
```
with `generateCompetency` generating that line item element

File: pages/web-literacy/WebLiteracyPage.jsx [hunk 6]
```diff
@@ -98,53 +49,108 @@ var Activity = React.createClass({
...
+        <div><b>Web Literacy Skills:</b> {makeLinksFromWebLitSkills(this.props.webLitSkills)}</div>
+        <div><b>21C Skills:</b> {makeLinksFrom21CSkills(this.props.skills)}</div>
       </Illustration>
     );
   }
 });
 
-module.exports = React.createClass({
+var WebLitPage = React.createClass({
+  mixins: [History],
```
Comment (Pomax) [inline pages/web-literacy/WebLiteracyPage.jsx]: what does this do?

File: pages/web-literacy/WebLiteracyPage.jsx [hunk 7]
```diff
@@ -98,53 +49,108 @@ var Activity = React.createClass({
...
+    }
+
+    return weblitdata[selectedVerb][selectedWebLitSkill].indexOf(cat) !== -1;
+  },
   renderActivities: function() {
     var selectedTopic = this.state.topic;
-    var selectedCompetency = this.state.competency;
-    var hasCompetencyIn = this.hasCompetencyIn;
-    var hasMatchingCompetencyIn = this.hasMatchingCompetencyIn;
+    var selectedWebLitSkill = this.state.webLitSkill;
```
Comment (Pomax) [inline pages/web-literacy/WebLiteracyPage.jsx]: hmmm... do we really need to do this? It does make L148 and L153 shorter, but not by a whole lot

Comment (Pomax) [inline pages/web-literacy/WebLiteracyPage.jsx]: I'm tempted to say let's not do this, as it hides what these things are a bit further on in the code. You'd have to read through the entire function to find out that they were just `this.[...]` aliases.

**Comment by Pomax** [comment]
We may want to run the `svg` files through [svgo](http://npmjs.com/package/svgo) before considering this PR ready to merge, too.

File: pages/web-literacy/MakeLinksFromWebLitSkills.jsx (hunk 4 above)
Comment (ScottDowne) [inline pages/web-literacy/MakeLinksFromWebLitSkills.jsx]: Cool!

File: pages/web-literacy/WebLiteracyPage.jsx (hunk 6 above)
Comment (ScottDowne) [inline pages/web-literacy/WebLiteracyPage.jsx]: Allows `this.history.pushState(null, url);` from inside `onMapToggle`

Which allows a button to navigate to a route.

It's the react-router 1.x way to do this that I found in the docs, after googling for... too long.

Comment (Pomax) [inline pages/web-literacy/WebLiteracyPage.jsx]: interesting...

Comment (ScottDowne) [inline pages/web-literacy/WebLiteracyPage.jsx]: I'm open to other ideas here.

File: less/pages/web-literacy.less (hunk 1 above)
Comment (ScottDowne) [inline less/pages/web-literacy.less]: Turns out these colours might change between now and tomorrow. I'll move them to variables.less for sure, not sure if I'll bother doing anything else to them until they are hardened.

File: pages/web-literacy/Skills.jsx (hunk 5 above)
Comment (ScottDowne) [inline pages/web-literacy/Skills.jsx]: I want to put both maps into a render function.

Comment (ScottDowne) [inline pages/web-literacy/Skills.jsx]: Separate render functions.

File: pages/web-literacy/WebLiteracyPage.jsx (hunk 6 above)
Comment (Pomax) [inline pages/web-literacy/WebLiteracyPage.jsx]: I'd rather do that in followup

**Pull request merged**